src/compiler/crystal/semantic/default_arguments.cr

```diff
@@ -104,6 +104,7 @@ class Crystal::Def
     expansion.raises = raises?
     expansion.free_vars = free_vars
     expansion.annotations = annotations
+    expansion.special_vars = special_vars
```

Comment:
@asterite do you recall if this was a fix of a bug that surfaced in the interpreter or something else?

Reply:
Special vars are done a bit differently in the interpreter compared to LLVM, and without this that way doesn't work. I think! You can try commenting this and checking if special vars (`$?` and so on) still work.